src/importlinter/adapters/filesystem.py

```diff
@@ -12,8 +12,17 @@ def join(self, *components: str) -> str:
         return os.path.join(*components)
 
     def read(self, file_name: str) -> str:
+        """
+        Return all non "\n" lines as a string while preserving formatting.
+        """
+        cleaned_file_str = ""
         with open(file_name) as file:
-            return file.read()
+            for line in file.readlines():
+                if line == '\n':
```

Comment:
I hear there's a way to instruct Python to look for whatever the system setting is for the end line character. Will come back to this when the test is written.